unison-cli/src/Unison/Codebase/Editor/HandleInput/Update.hs

```diff
@@ -189,12 +201,324 @@ handleUpdate input optionalPatch requestedNames = do
         & Path.resolve @_ @_ @Path.Absolute currentPath'
         & tShow
 
+getSlurpResultForUpdate :: Set Name -> Cli r (SlurpResult Symbol)
+getSlurpResultForUpdate requestedNames = do
+  (slurp, termRefToNames, nameToTermRefs) <- do
+    slurpCheckNames <- Branch.toNames <$> Cli.getCurrentBranch0
+
+    let slurp :: TypecheckedUnisonFile Symbol Ann -> SlurpResult Symbol
+        slurp file =
+          Slurp.slurpFile file (Set.map Name.toVar requestedNames) Slurp.UpdateOp slurpCheckNames
+
+    let termRefToNames :: TermReferenceId -> Set Symbol
+        termRefToNames =
+          Set.map Name.toVar . Names.namesForReferent slurpCheckNames . Referent.fromTermReferenceId
+
+    let nameToTermRefs :: Symbol -> Set TermReference
+        nameToTermRefs = Names.refTermsNamed slurpCheckNames . Name.unsafeFromVar
+
+    pure (slurp, termRefToNames, nameToTermRefs)
+
+  slurp1 <- do
+    Cli.Env {codebase} <- ask
+    unisonFile0 <- Cli.expectLatestTypecheckedFile
+
+    -- Here, we identify whether there are any "implicit terms", which are terms that should be brought into the latest
+    -- typechecked Unsion file and re-typechecked, which also re-discovers components.
```

Comment:
What does it mean to `re-discover` components? Is this sentence just saying that we find the components of all things being updated?